awscli/autoprompt/factory.py

```diff
@@ -430,39 +499,55 @@ def apply_transformation(self, text_input):
         return Transformation(fragments)
 
 
-class ToolbarHelpText:
-    def __init__(self, style=None, spacing=None):
+class HelpText:
```

Comment:
Ok I think it makes sense if we start refactoring this class and try to make it more generalized/reusable. Specifically, I'm thinking we make this class more of a factory class that returns help text based on some identifier for the component we are getting help text for. For example:

```python
class BaseHelpFormatter:
     def format(self):
          raise NotImplementedError('format')


class ToolbarHelpFormatter(BaseHelpFormatter):
    KEYS_TO_DESCRIPTION = {}
    def format(self):
         # Format string using spaces


class InputBufferToolbarHelpFormatter(ToolbarHelpFormatter):
    KEYS_TO_DESCRIPTION = {
         # Put the keys we care about here
    }


 # Add similar classes for the full help windows but make them delineated by newlines


class HelpTextGetter:
     def __init__(self, help_texts=None):
           if not help_texts:
                help_texts = self._get_default_help_texts()  # This just uses the formatter classes to build up the classes
           self._help_texts = help_texts  # type: Dict[str, str] where the key is the id and the value is the help text

     def get_help_text(self, help_id):
            # Returns the help text based on the id provided
```
Then getting help text to populate the buffers would be a matter of calling the `get_help_text()` function with the correct id. For example:
```
help_text_getter.get_help_text('input_buffer_tool_bar')
```

Reply:
Maybe we can move to another direction. It's just a bunch of strings so we can  just get rid of class and create a bunch of constants.

Reply:
Yeah the works too. The only reason I added the classes so we can reuse logic in building up these big menus.

Reply:
Looking over more of this conversation, another idea is that we could just implement our own prompt-toolkit container class representing a particular help window/buffer and just make the content constants of the class. I think that can help consolidate all of this logic as a well to a single class.